src/StreamJsonRpc/Reflection/MessageFormatterRpcMarshaledContextTracker.cs

```diff
@@ -87,6 +87,44 @@ private enum MarshalMode
             MarshallingRealObject = 1,
         }
 
+        internal static (JsonRpcProxyOptions ProxyOptions, JsonRpcTargetOptions TargetOptions)? GetMarshalOptionsForType(Type type)
+        {
+            Requires.NotNull(type, nameof(type));
+
+            foreach ((Type implicitlyMarshaledType, JsonRpcProxyOptions proxyOptions, JsonRpcTargetOptions targetOptions) in ImplicitlyMarshaledTypes)
+            {
+                if (implicitlyMarshaledType == type ||
+                    (implicitlyMarshaledType.IsGenericTypeDefinition &&
+                     type.IsConstructedGenericType &&
+                     implicitlyMarshaledType == type.GetGenericTypeDefinition()))
+                {
+                    return (proxyOptions, targetOptions);
+                }
+            }
+
+            if (type.IsInterface && type.GetCustomAttribute<RpcMarshalableAttribute>() is not null)
+            {
+                if (typeof(IDisposable).IsAssignableFrom(type) is false)
+                {
+                    throw new NotSupportedException(Resources.MarshalableInterfaceNotDisposable);
+                }
+
+                if (type.GetEvents().Any())
+                {
+                    throw new NotSupportedException(Resources.MarshalableInterfaceHasEvents);
+                }
+
+                if (type.GetProperties().Any())
+                {
+                    throw new NotSupportedException(Resources.MarshalableInterfaceHasProperties);
+                }
+
+                return (new JsonRpcProxyOptions(), new JsonRpcTargetOptions { NotifyClientOfEvents = false, DisposeOnDisconnect = true });
```

Comment:
@AArnott, what should we do for `DisposeOnDisconnect` for custom interfaces? I opted for `true` considering that we are forcing the interface to extend `IDisposable`, but I don't have a strong opinion on this.

Reply:
I think that's a good default.